spring-boot-modules/spring-boot-validations/src/main/java/com/baeldung/dto/BooleanObject.java

```diff
@@ -0,0 +1,55 @@
+package com.baeldung.dto;
+
+import javax.validation.constraints.AssertFalse;
+import javax.validation.constraints.AssertTrue;
+import javax.validation.constraints.NotNull;
+
+import com.baeldung.deserializer.BooleanDeserializer;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+
+public class BooleanObject {
+
+    @NotNull(message = "boolField cannot be null")
+    Boolean boolField;
+
+    @AssertTrue(message = "trueField must have true value")
+    Boolean trueField;
+
+    @AssertFalse(message = "falseField must have false value")
```

Comment:
Add: @NotNull(message = "falseField cannot be null")

Reply:
Changes implemented